packages/rsocket-types/src/RSocket.ts

```diff
@@ -87,28 +87,26 @@ export interface RSocket extends Closeable {
    * Fire and Forget interaction model of `ReactiveSocket`. The returned
    * Publisher resolves when the passed `payload` is successfully handled.
    */
-  fireAndForget(
-    payload: Payload,
-    responderStream: UnidirectionalStream
-  ): Cancellable;
+  fireAndForget(payload: Payload, responderStream: Subscriber): Cancellable;
 
   /**
    * Request-Response interaction model of `ReactiveSocket`. The returned
    * Publisher resolves with the response.
    */
   requestResponse(
     payload: Payload,
-    responderStream: UnidirectionalStream
-  ): Cancellable;
+    responderStream: Subscriber & ExtensionSubscriber
+  ): Cancellable & ExtensionSubscriber;
 
   /**
    * Request-Stream interaction model of `ReactiveSocket`. The returned
    * Publisher returns values representing the response(s).
    */
   requestStream(
     payload: Payload,
-    responderStream: UnidirectionalStream
-  ): Subscription;
+    initialRequestN: number,
```

Comment:
we need to have `requestN` since this is part of RequestStream Frame. So this is an inevitable part to make the request for stream or channel.

Also, this API is the API we have in the swift impl and is likely to be the standard for all the impl.  (https://github.com/rsocket/rsocket-swift/blob/main/Sources/RSocketCore/RSocket.swift)

Reply:
Thanks for the exaplanation. As we discussed in our call, we'll look to retain the functionality of sending `initialRequestN` on the initial `REQUEST_STREAM` frame, however, we may look to modify the subscriber API to accept tuple or other settings object type arguments in the future, either via helper or some other extension mechanism.